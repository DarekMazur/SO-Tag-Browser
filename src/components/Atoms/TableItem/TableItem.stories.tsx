import type { Meta, StoryObj } from '@storybook/react';
import { faker } from '@faker-js/faker';
import TableItem from './TableItem.tsx';
import TagLink from '../TagLink/TagLink.tsx';

const meta: Meta<typeof TableItem> = {
  tags: ['autodocs'],
  component: TableItem,
  title: 'Atoms/TableItem',
};

export default meta;

type Story = StoryObj<typeof TableItem>;

export const Default: Story = {
  argTypes: {
    label: {
      control: 'text',
      description: 'Item content',
      name: 'label',
    },
  },
  args: {
    label: faker.lorem.word(),
  },
};

export const WithIcon: Story = {
  args: {
    label: <TagLink to={faker.lorem.word()} />,
  },
};
