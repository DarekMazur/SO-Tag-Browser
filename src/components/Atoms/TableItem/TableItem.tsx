import { TableCell } from '@mui/material';
import { FC, ReactNode } from 'react';

interface ITableItemProps {
  label: string | ReactNode;
  align?: 'left' | 'center' | 'right';
}

const tableItems: FC<ITableItemProps> = ({ label, align }) => {
  return <TableCell align={align || 'left'}>{label}</TableCell>;
};

export default tableItems;
