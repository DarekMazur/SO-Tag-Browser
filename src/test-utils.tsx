// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { ReactNode } from 'react';
import AppProviders from './providers/AppProviders.tsx';

const AllTheProviders = ({ children }: { children: ReactNode }) => {
  return <AppProviders>{children}</AppProviders>;
};

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
const customRender = (ui, options?) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react';

export { customRender as render };
