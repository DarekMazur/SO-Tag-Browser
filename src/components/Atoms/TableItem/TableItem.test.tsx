import { render, screen } from '../../../test-utils';
import TableItem from './TableItem.tsx';

describe('Table Item', () => {
  it('Renders the component', () => {
    const { container } = render(<TableItem label="example" />);
    expect(container).toMatchSnapshot();
  });
  it('Renders with label', () => {
    render(<TableItem label="example" />);
    expect(screen.getByText('example')).toBeInTheDocument();
  });
  it('Renders with style', () => {
    render(<TableItem label="example" align="right" />);
    expect(screen.getByText('example')).toHaveStyle({ textAlign: 'right' });
  });
});
