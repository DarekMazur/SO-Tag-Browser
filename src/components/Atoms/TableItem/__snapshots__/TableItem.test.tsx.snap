// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Table Item > Renders the component 1`] = `
<div>
  <td
    class="MuiTableCell-root MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1alr223-MuiTableCell-root"
  >
    example
  </td>
</div>
`;
